app/src/main/java/foxsay/ru/loftmoney/AddItemActivity.java

```diff
@@ -20,26 +17,13 @@ protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_add_item);
 
-        name = (EditText) findViewById(R.id.add_item_name);
-        price = (EditText) findViewById(R.id.add_item_price);
-        addBtn = (Button) findViewById(R.id.add_item_button);
+        name = findViewById(R.id.add_item_name);
+        price = findViewById(R.id.add_item_price);
+        addBtn = findViewById(R.id.add_item_button);
 
-        name.addTextChangedListener(new TextWatcher() {
-            @Override
-            public void beforeTextChanged(CharSequence s, int start, int count, int after) {
-
-            }
-
-            @Override
-            public void onTextChanged(CharSequence s, int start, int before, int count) {
-
-            }
-
-            @Override
-            public void afterTextChanged(Editable s) {
-                addBtn.setEnabled(!TextUtils.isEmpty(s));
-            }
-        });
+        name.addTextChangedListener(new EditTextWatcher(price, addBtn));
```

Comment:
try to make it easier. Without creating a new class

Reply:
made changes please check the code